File: synthetix/runs/20260504_083830_80288286/qa/generated_code/authservice/main.py
from fastapi import FastAPI
from handlers import router
import os

app = FastAPI()

app.include_router(router)

@app.get("/health")
def health_check():
    return {"status": "healthy"}

@app.get("/ready")
def readiness_check():
    return {"status": "ready"}

if __name__ == "__main__":
    import uvicorn
    port = int(os.getenv("PORT", 8080))
    uvicorn.run(app, host="0.0.0.0", port=port)

File: synthetix/runs/20260504_083830_80288286/qa/generated_code/authservice/main.go
package main

import (
	"log"
	"net/http"
	"os"

	"github.com/go-chi/chi/v5"
	"github.com/go-chi/chi/v5/middleware"
)

func main() {
	r := chi.NewRouter()
	r.Use(middleware.Logger)

	r.Get("/health", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte(`{"status": "healthy"}`))
	})

	r.Get("/ready", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte(`{"status": "ready"}`))
	})

	r.Mount("/api/auth", authRouter())

	port := os.Getenv("PORT")
	if port == "" {
		port = "8080"
	}

	log.Printf("Starting server on :%s", port)
	if err := http.ListenAndServe(":"+port, r); err != nil {
		log.Fatalf("Error starting server: %v", err)
	}
}

func authRouter() http.Handler {
	r := chi.NewRouter()

	r.Post("/apiKey", apiKeyHandler)
	r.Post("/register", registerHandler)
	r.Post("/login", loginHandler)
	r.Post("/refresh", refreshHandler)
	r.Post("/logout", logoutHandler)
	r.Post("/resend-verification", resendVerificationHandler)
	r.Post("/verify", verifyHandler)
	r.Post("/forgot-password", forgotPasswordHandler)

	return r
}